05.wc/wc.rb

```diff
@@ -0,0 +1,76 @@
+# frozen_string_literal: true
+
+require 'optparse'
+
+def parse_options
+  options = { lines: false, words: false, bytes: false }
+  OptionParser.new do |opts|
+    opts.banner = 'Usage: wc.rb [options] [file ...]'
+    opts.on('-l', 'Count lines') { options[:lines] = true }
+    opts.on('-w', 'Count words') { options[:words] = true }
+    opts.on('-c', 'Count bytes') { options[:bytes] = true }
+  end.parse!
+  options.transform_values! { |_| true } unless options.values.any?
+  options
+end
+
+def count_lines(content)
+  content.count("\n")
+end
+
+def count_words(content)
+  content.split(/\s+/).reject(&:empty?).count
+end
+
+def count_bytes(content)
+  content.bytesize
+end
+
+def output_results(counts, options, filename = nil, total: false)
+  output = []
+  output << format('%8d', counts[:lines]) if options[:lines]
+  output << format('%8d', counts[:words]) if options[:words]
+  output << format('%8d', counts[:bytes]) if options[:bytes]
+  output << " #{filename}" if filename
+  output << ' total' if total && filename.nil?
+
+  puts output.join(' ')
+end
+
+def handle_multiple_files(files, options)
+  total_counts = files.map { |file| process_single_file(file, options) }
+
+  return unless files.size > 1
+
+  total_lines, total_words, total_bytes = total_counts.map(&:values).transpose.map(&:sum)
+  output_results({ lines: total_lines, words: total_words, bytes: total_bytes }, options, nil, total: true)
```

Comment:
```suggestion
  output_results({ lines: total_lines, words: total_words, bytes: total_bytes }, options, total: true)
```

filenameはデフォルト値が指定されているので引数を省略できますね。
あと、第1引数は先に変数に入れておいた方が読みやすいと思います

```suggestion
  hoge = { lines: total_lines, words: total_words, bytes: total_bytes }
  output_results(hoge, options, total: true)
```